client/src/components/DisplayTable.js

```diff
@@ -16,17 +16,18 @@ export class DisplayTable extends React.Component {
 
     const { rows } = props;
 
-    const { sort_values, search_values } = _.first(rows);
+    const { sort_values } = _.first(rows);
 
     const sort_by = _.chain(sort_values)
       .keys()
       .first()
       .value();
 
-    const searches = _.mapValues(
-      search_values,
-      () => "",
-    );
+    const searches = _.chain(rows)
+      .map( row => _.keys(row.search_values) )
+      .union().keyBy()
+      .mapValues( () => "" )
+      .value();
```

Comment:
Not clear what this is doing. Is it equivalent to this?
```
const searches = _.chain(rows)
  .flatMap( 
    ({search_values}) => _.chain(search_values)
      .keys()
      .map( (search_key) => [search_key, ""] )
      .value()
  )
  .fromPairs()  // left out the uniq-ing step, happens implicitly here since the resulting object will only have one instance of each key
  .value();
```

Reply:
Basically it needs to get all the search keys from all rows (it was assuming the first row had all the data) and setting up the `searches` object like so:

```
searches = {
  key1: "",
  key2: "",
  key3: "",
}
```
I think your code is equivalent, yeah.